router/batchrouter/asyncdestinationmanager/salesforce-bulk-upload/augmenter/augmenter.go

```diff
@@ -0,0 +1,65 @@
+package augmenter
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/samber/lo"
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-server/services/oauth/v2/common"
+	oauthv2extensions "github.com/rudderlabs/rudder-server/services/oauth/v2/extensions"
+)
+
+type requestAugmenter struct{}
+
+func NewRequestAugmenter() oauthv2extensions.Augmenter {
+	return &requestAugmenter{}
+}
+
+// Custom augmenter for Salesforce which sets token to Authorization header and instance URL to the request URL
+func (s *requestAugmenter) Augment(r *http.Request, body []byte, secret json.RawMessage) error {
+	if secret == nil {
+		return errors.New("secret is nil")
+	}
+	accessToken := gjson.GetBytes(secret, "access_token").String()
+	if accessToken == "" {
+		return errors.New("access token is empty")
+	}
+	instanceURL := gjson.GetBytes(secret, "instance_url").String()
+	if instanceURL == "" {
+		return errors.New("instance URL is empty")
+	}
+	instanceURL = strings.Replace(instanceURL, "https://", "", 1)
+	// format -> Authorization : OAuth <accessToken>
+	r.Header.Set("Authorization", fmt.Sprintf("Bearer %s", accessToken))
+	r.URL.Host = instanceURL
+	r.Body = io.NopCloser(bytes.NewReader(body))
+	return nil
+}
+
+// GetAuthErrorCategoryForSalesforce returns the error category for Salesforce authentication errors
+
+func GetAuthErrorCategoryForSalesforce(responseBody []byte) (string, error) {
```

Comment:
We are returning nil in both the scenarios, do we even need to send error?
```suggestion
func GetAuthErrorCategoryForSalesforce(responseBody []byte) string {
```

Reply:
Later while initialising client we can do something like this.
```GO
client := oauthv2httpclient.NewOAuthHttpClient(
		originalHttpClient,
		oauthv2common.RudderFlowDelivery,
		&cache,
		backendConfig,
		func(responseBody []byte) (string, error) {
			return augmenter.GetAuthErrorCategoryForSalesforce(responseBody), nil
		},
		optionalArgs,
	)
```